test/suites/container_devices.sh

```diff
@@ -172,7 +172,51 @@ test_container_devices_nic() {
   # Test removing non-veth macvlan from running container.
   lxc config device remove "${ct_name}" eth1
 
-  # Cleanup.
+  # Cleanup veth/macvlan checks.
+  lxc delete "${ct_name}" -f
+  ip link delete "${ct_name}" type dummy
+
+  # Test ipvlan support to offline container (hot plugging not supported).
+  echo "Testing IPVLAN support..."
+  ip link add "${ct_name}" type dummy
+
+  # Check that starting IPVLAN container.
```

Comment:
So this test is going to be problematic because if liblxc is lacking IPVLAN, this will all fail and Linux distributions which run our testsuite on their packages do not usually have the current master of liblxc.

I'm failing to come up with a cleaner way, so maybe have this deal properly with the error for missing lxc extension and move on?

Reply:
@stgraber how about adding a test go binary to the LXD test suite that loads liblxc using go-lxc and then allows us to query for supported LXC extensions, which we can then execute from the test suite?

Reply:
@stgraber actually, what about dynamically altering the output from `lxc info` in terms of the API extensions by modifying this file https://github.com/lxc/lxd/blob/master/shared/version/api.go#L23 to add the IPVLAN LXD functionality based on whether the LXC functionality is available (via go-lxc)? Then the tests can run `lxc info` to figure out whether they need to test that feature first.

Reply:
Hmm, I was considering doing something similar for the seccomp one in liblxc but I think this will cause us problems. We should really keep things separate there.

We could possibly add a `driver_features` map similar to `kernel_features` to report support for those extensions we care about though.

Reply:
@stgraber OK sounds good I will look at adding that.

Reply:
@stgraber this has been added now. I've also split apart the tests for different nic types into different files as they were getting unwieldy.